instaloader/structures.py

```diff
@@ -408,6 +408,21 @@ def get_sidecar_nodes(self, start=0, end=-1) -> Iterator[PostSidecarNode]:
                     yield PostSidecarNode(is_video=is_video, display_url=display_url,
                                           video_url=node['video_url'] if is_video else None)
 
+    def get_translated_caption(self) -> Optional[str]:
+        """Retrieve translation of the caption.
+
+        .. versionadded:: 4.10"""
+        if self.caption and self._context.iphone_support and self._context.is_logged_in:
+            try:
+                pk = self._iphone_struct['caption']['pk']
+                data = self._context.get_iphone_json(path=f'api/v1/language/bulk_translate/?comment_ids={pk}',
+                                                    params={})
+                self._node["translated_caption"] = data['comment_translations'][0]['translation']
+                return data['comment_translations'][0]['translation']
+            except (InstaloaderException, KeyError, IndexError) as err:
+                self._context.error(f"Unable to get translation: {err}")
```

Comment:
`pk` should be generically passed as a parameter, the function should be moved in the context since it can serve more than only the Post, the translation should not be stored in the node, rather the translation strings could be cashed with a dict by `pk`.